python/lsst/ip/diffim/modelPsfMatch.py

```diff
@@ -372,15 +394,42 @@ def _buildCellSet(self, exposure, referencePsfModel):
                 widthList.append(widthS)
                 heightList.append(heightS)
 
-        lenMax = max(max(heightList), max(widthList))
-        lenMin = min(min(heightList), min(widthList))
+        psfSize = max(max(heightList), max(widthList))
 
-        lenPsfScience = lenMax if self.config.padPsf else lenMin
-        dimenS = afwGeom.Extent2I(lenPsfScience, lenPsfScience)
+        if self.config.doAutoPadPsf:
+            minPsfSize = nextOddInteger(self.kConfig.kernelSize*self.config.autoPadPsfTo)
+            paddingPix = max(0, minPsfSize - psfSize)
+        else:
+            if self.config.padPsfBy % 2 != 0:
+                raise ValueError("Config padPsfBy (%i pixels) must be even number." %
```

Comment:
Is it worth just moving to the next even number rather than throwing an error? Or maybe being explicit is good, like you have it. If the algorithm must have an even to work, perhaps move to the next even number and put out a warning? What you have is fine, these are just some things for you to consider.